### PR TITLE
[ENG-300] Explorer multi-select

### PR DESCRIPTION
**Pull request by niikeec**
This PR introduces multi-select.

What's New:

Multi-Select across all layouts: 

multi-select

The inspector now displays relevant data whether you're inspecting a single item or multiples:

inspector

[Context menu support for multi-select](https://github.com/spacedriveapp/spacedrive/pull/1187)

### CODE REVIEW

**Comment by jamiepine** [comment]
Sooo excited for this PR, trying to run it but getting this
Screenshot 2023-08-12 at 12 47 25 AM

**Comment by jamiepine** [comment]
i look at this PR when i'm sad to feel better

File: interface/app/$libraryId/Explorer/ContextMenu/Object/utils.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { useMemo } from 'react';
```
Comment (utkubakir) [inline interface/app/$libraryId/Explorer/ContextMenu/Object/utils.ts:1]: Can move this to `@sd/client`

**Comment by Brendonovich** [comment]
> i look at this PR when i'm sad to make me feel better
@jamiepine do you need a friend

File: interface/app/$libraryId/Explorer/RevealInNativeExplorer.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	const osFileBrowserName = lookup[os] ?? 'file manager';
+
+	return (
+		<ContextMenu.Item
+			label={`Reveal in ${osFileBrowserName}`}
+			keybind={keybind([ModifierKeys.Control], ['Y'])}
+			onClick={() => revealItems(library.uuid, props.items)}
+		/>
+	);
+};
```
Comment (utkubakir) [inline interface/app/$libraryId/Explorer/RevealInNativeExplorer.tsx:33]: Shouldn't this be also a conditional context item?

File: interface/app/$libraryId/Explorer/View/index.tsx [hunk 3]
```diff
@@ -203,50 +188,44 @@ export const EmptyNotice = ({
...
 
-	const selectedItem = useMemo(
-		() =>
-			items?.find(
-				(item) => item.item.id === (Array.isArray(selected) ? selected[0] : selected)
-			),
-		[items, selected]
-	);
+	/// TODO: This ain't right!
+	const selectedItem = [...explorer.selectedItems][0];
```
Comment (utkubakir) [inline interface/app/$libraryId/Explorer/View/index.tsx:212]: ?

File: interface/app/$libraryId/Explorer/RevealInNativeExplorer.tsx (hunk 2 above)
Comment (Brendonovich) [reply interface/app/$libraryId/Explorer/RevealInNativeExplorer.tsx:33]: Nah, this is a base component that's used in a couple places, one of them being a conditional:
https://github.com/spacedriveapp/spacedrive/blob/f556fd2ef833abca310bfa07815148a1e0f20643/interface/app/%24libraryId/Explorer/ContextMenu/SharedItems.tsx#L73
https://github.com/spacedriveapp/spacedrive/blob/f556fd2ef833abca310bfa07815148a1e0f20643/interface/app/%24libraryId/Explorer/ParentContextMenu.tsx#L107

**Comment by utkubakir** [comment]
this pr is magical

File: interface/app/$libraryId/Explorer/View/index.tsx (hunk 3 above)
Comment (Brendonovich) [reply interface/app/$libraryId/Explorer/View/index.tsx:212]: Ah whoops, never got to implementing multi-item support for the key handlers.

File: interface/app/$libraryId/Explorer/ContextMenu/FilePath/Items.tsx [hunk 4]
```diff
@@ -128,102 +141,126 @@ export const Crypto = (_: FilePathProps) => {
...
-									.mutateAsync(filePath.object_id)
+									.mutateAsync(selectedFilePaths[0].object_id)
 									.catch(console.error);
 
 							try {
-								await openFilePath(library.uuid, [filePath.id]);
+								await openFilePaths(
+									library.uuid,
+									selectedFilePaths.map((p) => p.id)
+								);
```
Comment (utkubakir) [inline interface/app/$libraryId/Explorer/ContextMenu/FilePath/Items.tsx:263]: found an another TODO

**Review by utkubakir — APPROVED** [review]
LGTM

**Review by Raghav-45 — APPROVED** [review]

**Review by jamiepine — APPROVED** [review]
without further adieu

**Pull request merged**